### PR TITLE
import dnxcore50

### LINKED ISSUE #2499: Add dnxcore50 imports back to dotnet new for RC2

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Fixes #2499 

Adds dnxcore50 to C# dotnet-new template
Adds Unit Test validating that a sample legacy portable NuGet package restores in a `dotnet new` project without user intervention.

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2548)

### CODE REVIEW

File: test/dotnet-new.Tests/project.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "dotnet": {
+      "target": "project"
+    },
+    "xunit": "2.1.0",
+    "dotnet-test-xunit": "1.0.0-dev-140469-38"
+  },
+  "frameworks": {
+    "netcoreapp1.0": {
+      "imports": [
+        "netstandardapp1.5",
```
Comment (eerhardt) [inline test/dotnet-new.Tests/project.json]: Do you need netstandardapp?

File: src/dotnet/commands/dotnet-new/CSharp_Console/project.json.pretemplate [hunk 2]
```diff
@@ -10,6 +10,8 @@
     }
   },
   "frameworks": {
-    "netcoreapp1.0": {}
+    "netcoreapp1.0": {
+      "imports": "dnxcore50"
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-new/CSharp_Console/project.json.pretemplate]: Does F# need this?

File: test/dotnet-new.Tests/GivenThatIWantANewCSApp.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        [Fact]
+        public void When_NewtonsoftJson_dependency_added_Then_project_restores_and_runs()
+        {
+            var rootPath = Temp.CreateDirectory().Path;
+            var projectJsonFile = Path.Combine(rootPath, "project.json");
+
+            new TestCommand("dotnet") { WorkingDirectory = rootPath }
+                .Execute("new");
+            
+            AddProjectJsonDependency(projectJsonFile, "Newtonsoft.Json", "7.0.1");
```
Comment (eerhardt) [inline test/dotnet-new.Tests/GivenThatIWantANewCSApp.cs]: I think you should add code to the app that uses Newtonsoft. Last I tried you need an extra dependency to System.Runtime.Serialization.Primitives in order to actually use Newtonsoft.

File: test/dotnet-new.Tests/project.json (hunk 1 above)
Comment (Petermarcu) [inline test/dotnet-new.Tests/project.json]: Do we still have a limitation somewhere requiring this to be netstandardapp? When we move to 1.6, it won't be there.

File: src/dotnet/commands/dotnet-new/CSharp_Console/project.json.pretemplate (hunk 2 above)
Comment (brthor) [inline src/dotnet/commands/dotnet-new/CSharp_Console/project.json.pretemplate]: f# already has it :)

File: .gitignore [hunk 4]
```diff
@@ -142,6 +142,8 @@ $tf/
 _ReSharper*/
 *.[Rr]e[Ss]harper
 *.DotSettings.user
+.idea/
```
Comment (brthor) [inline .gitignore]: ?

File: test/dotnet-new.Tests/GivenThatIWantANewCSApp.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+using System.Linq;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using FluentAssertions;
+
+namespace Microsoft.DotNet.Tests
+{
+    public class GivenThatIWantANewCSApp : TestBase
```
Comment (brthor) [inline test/dotnet-new.Tests/GivenThatIWantANewCSApp.cs]: 👍 preferring these style of tests lately

File: test/dotnet-new.Tests/project.json (hunk 1 above)
Comment (brthor) [inline test/dotnet-new.Tests/project.json]: I don't think we should need this

Comment (piotrpMSFT) [inline test/dotnet-new.Tests/project.json]: fixing this

File: .gitignore (hunk 4 above)
Comment (piotrpMSFT) [inline .gitignore]: Tried using JetBrains Rider with CLI on Mac. Adding ignores for their artifacts.

Comment (brthor) [inline .gitignore]: oh sweet does it work with CLI now?

**Comment by brthor** [comment]
I think removing the netstandardapp import is the only concern I have.

Once removed :shipit:

File: .gitignore (hunk 4 above)
Comment (piotrpMSFT) [inline .gitignore]: Nope. My experiment failed.

**Pull request merged**